app/scripts/app-init.js

```diff
@@ -132,12 +132,15 @@ chrome.runtime.onMessage.addListener(() => {
  * MAIN world injection does not work properly via manifest
  * https://bugs.chromium.org/p/chromium/issues/detail?id=634381
  */
-chrome.scripting.registerContentScripts([
-  {
-    id: 'inpage',
-    matches: ['file://*/*', 'http://*/*', 'https://*/*'],
-    js: ['inpage.js'],
-    runAt: 'document_start',
-    world: 'MAIN',
-  },
-]);
+// eslint-disable-next-line no-undef
+self.addEventListener('install', () => {
```

Comment:
this caused a race condition on my macOS x chrome

Repro steps:
1. refresh extension from `chrome://extensions/` 
2. open or refresh dapp
3. observe provider is missing and dapp does not work 
<img width="544" alt="Screenshot 2022-09-24 at 1 51 41 AM" src="https://user-images.githubusercontent.com/20778143/192080610-0cb112a8-36d3-4c9e-94ea-96b058c494ee.png">

I think, for now,  it would be better to catch the error instead and prioritize other work. The duplicate script error will drop the attempt to register a new inpage script and the existing inpage script will still be loaded. This should not cause any visible errors with our existing logic. 

I created a commit here to catch the error, for now:
https://github.com/MetaMask/metamask-extension/pull/15939/commits/26ba321a4af0858261518fd9649768050eeb9c62

Reply:
Good catch @digiwand , let me revert this change.

Reply:
@jpuri  

I created a separate PR and issue ticket for the "Duplicate script ID 'inpage'" bug:
Issue: https://github.com/MetaMask/metamask-extension/issues/15971
PR: https://github.com/MetaMask/metamask-extension/pull/15973